Tests/UnitTests/Core/Geometry/GeometryIdGeneratorsTests.cpp

```diff
@@ -0,0 +1,267 @@
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <boost/test/unit_test.hpp>
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Geometry/CuboidVolumeBounds.hpp"
+#include "Acts/Geometry/DetectorElementBase.hpp"
+#include "Acts/Geometry/DetectorVolume.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Geometry/NavigationState.hpp"
+#include "Acts/Geometry/detail/DetectorVolumeUpdators.hpp"
+#include "Acts/Geometry/detail/GeometryIdGenerators.hpp"
+#include "Acts/Geometry/detail/PortalGenerators.hpp"
+#include "Acts/Geometry/detail/PortalHelper.hpp"
+#include "Acts/Geometry/detail/SurfaceCandidatesUpdators.hpp"
+#include "Acts/Surfaces/PlaneSurface.hpp"
+#include "Acts/Surfaces/RectangleBounds.hpp"
+#include "Acts/Utilities/Enumerate.hpp"
+
+#include <array>
+#include <memory>
+#include <vector>
+
+namespace Acts {
+
+/// A mockup detector element
+class TestDetectorElement : public DetectorElementBase {
+ public:
+  /// @brief  Constructor of a dummy test element
+  TestDetectorElement(const Transform3& trf = Transform3::Identity())
+      : m_transform(trf) {
+    m_surface = Surface::makeShared<PlaneSurface>(
+        std::make_shared<RectangleBounds>(10., 10.), *this);
+  }
+
+  /// Return the transform for the Element proxy mechanism
+  ///
+  /// @param gctx The current geometry context object, e.g. alignment
+  const Transform3& transform(const GeometryContext& /*gctx*/) const final {
+    return m_transform;
+  };
+
+  /// Return surface representation const access
+  const Surface& surface() const final { return *(m_surface.get()); }
+
+  /// Return surface representation, non- const access
+  Surface& surface() { return *(m_surface.get()); }
+
+  /// Returns the thickness of the module
+  /// @return double that indicates the thickness of the module
+  double thickness() const final { return 1.; }
+
+ private:
+  Transform3 m_transform = Transform3::Identity();
+  std::shared_ptr<PlaneSurface> m_surface = nullptr;
+};
+
+}  // namespace Acts
+
+Acts::GeometryContext tContext;
+
+BOOST_AUTO_TEST_SUITE(Experimental)
+
+namespace {
+
+/// @brief  This generates a vector of mockup volumes
+///
+///  | A | B | C | D  | E | F |
+///  |   | s |   | sp | p | v |
+///
+/// A -> empty volume
+/// B -> volume with sensitive surface
+/// C -> empty volume
+/// D -> volume with sensitive and passive surface
+/// E -> volume with passive surface
+/// F -> volume with sub volume
+///
+/// @return an array of 6 volumes with different content
+std::array<std::shared_ptr<Acts::Experimental::DetectorVolume>, 6u>
+mockupVolumes(std::vector<std::shared_ptr<Acts::TestDetectorElement>>& store) {
+  auto portals = Acts::Experimental::detail::defaultPortalGenerator();
+
+  Acts::ActsScalar hLength = 100.;
+
+  std::array<std::shared_ptr<Acts::Experimental::DetectorVolume>, 6u> mVolumes =
+      {nullptr, nullptr, nullptr, nullptr, nullptr, nullptr};
```

Comment:
```suggestion
  std::array<std::shared_ptr<Acts::Experimental::DetectorVolume>, 6u> mVolumes;
```

should be equivalent I think